### PR TITLE
[Ming-Omni] Support Ming-Omni ASR

### PR DESCRIPTION
**Pull request by yuan-luo**
## Motivation

The basic Ming-Omni audio-encoder has been supported. This PR is to resolve one point:

EncoderInputPreparer.prepare() handles tensors in two ways during batching:

cat_keys → torch.cat(tensors, dim=0) — preserves the original dimensions

Other tensors → torch.stack(tensors) — adds a new batch dimension

The problem is that audio_feats already has a batch dimension when it comes out of the preprocessor: [1, 300, 128].

Without the fix (taking the torch.stack path):
```
torch.stack([tensor of shape [1, 300, 128]])  →  [1, 1, 300, 128]  # 4D, an extra dim
```
Then MingAudioEncoder.forward()'s conv1d receives a 4D input and raises:
```
Expected 2D or 3D input to conv1d, but got input of size: [1, 300, 1, 128]
```
With the fix (taking the torch.cat path):
```
torch.cat([tensor of shape [1, 300, 128]], dim=0)  →  [1, 300, 128]  # 3D, correct
```
Fundamentally, image tensors like pixel_values and image_grid_thw are already in cat_keys (they have the same situation,  come with their own batch dimension), but Ming's audio tensor is newly added. No audio input had ever gone through this path before, which is why it was missing.

With the fix, Ming-Omni ASR is working correctly. Added a unit test to verify.

```
python tests/test_ming_asr.py --url http://localhost:8000 --audio /sgl-workspace/sglang-omni-dev/tests/data/query_to_cars.wav --benchmark --num-runs 5

2026-04-20 10:11:57,096 INFO Server OK: http://localhost:8000
2026-04-20 10:11:57,096 INFO Testing 'transcribe'...
2026-04-20 10:11:57,897 INFO Testing 'transcribe_zh'...
2026-04-20 10:12:00,178 INFO Testing 'understand'...
2026-04-20 10:12:06,082 INFO Testing 'summarize'...
2026-04-20 10:12:09,098 INFO Warmup...
2026-04-20 10:12:10,562 INFO   Run 1/5: 0.742s
2026-04-20 10:12:11,308 INFO   Run 2/5: 0.746s
2026-04-20 10:12:12,093 INFO   Run 3/5: 0.785s
2026-04-20 10:12:12,842 INFO   Run 4/5: 0.749s
2026-04-20 10:12:13,588 INFO   Run 5/5: 0.747s
2026-04-20 10:12:13,595 INFO Results saved to /tmp/asr_test_results.json

================================================================================
                       MING-OMNI ASR — FUNCTIONAL TESTS
================================================================================
 Audio: /sgl-workspace/sglang-omni-dev/tests/data/query_to_cars.wav

 [PASS] transcribe (0.80s)
        Prompt: Please transcribe this audio.
        Output: hello this is a test of the mingomnitalker model

 [PASS] transcribe_zh (2.28s)
        Prompt: 请逐字转写这段音频。
        Output: <question>Hello, this is a test of the mingomnitalker model. How</question><answer>Hello, this is a
test of the mingomnitalker model. How</answer>

 [PASS] understand (5.90s)
        Prompt: What language is spoken in this audio? What is the speaker talking about?
        Output: <question>Hello, this is a test of the mingomnitalker model. How</question><answer>The audio clip
contains a male voice speaking in English. The speaker is testing the Mingomnitalker model, which is a

 [PASS] summarize (1.03s)
        Prompt: Summarize the content of this audio in one sentence.
        Output: This audio clip features a man delivering a spoken message, with the content remaining undisclosed.

 Result: 4/4 tests passed
================================================================================

================================================================================
                    MING-OMNI ASR — PERFORMANCE BENCHMARK
================================================================================
 Audio: /sgl-workspace/sglang-omni-dev/tests/data/query_to_cars.wav
 Runs: 5

 Audio duration: 4.62s

 Metric                              Value
 ------------------------------------------
 Latency Mean                        0.754 s
 Latency Median                      0.747 s
 Latency P95                         0.778 s
 Latency Std                         0.016 s
 Audio Duration                       4.62 s
 RTF (lower=faster)                  0.163
 Real-time factor               6.1x real-time
================================================================================
```

## Checklist

- [ ] Format your code according with pre-commit.
- [ ] Add unit tests.
- [ ] Update documentation / docstrings / example tutorials as needed.
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed.
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.

### CODE REVIEW

**Review by FrankLeeeee — APPROVED** [review]

**Comment by FrankLeeeee** [comment]
LGTM, waiting for CI to pass
